src/graph.c

```diff
@@ -229,15 +229,17 @@ static void check_reachability(meshlink_handle_t *mesh) {
 
 			timeout_del(&mesh->loop, &n->mtutimeout);
 
-			update_node_status(mesh, n);
-
-			if(!n->status.reachable) {
-				update_node_udp(mesh, n, NULL);
-				n->status.broadcast = false;
-				n->options = 0;
-			} else if(n->connection) {
-				if(n->connection->outgoing) {
-					send_req_key(mesh, n);
+			if(!n->status.blacklisted) {
```

Comment:
I think you should only make `update_node_status()` part of this `if`-statement, the rest is bookkeeping for MeshLink internally, and should always be done, just to be on the safe side.